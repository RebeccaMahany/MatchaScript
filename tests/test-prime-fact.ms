function void primeNumberChecker(int a) {
	print("Current number:");
	print(a);
	int counter = 2;
	int prime = 1;
	int current_a = a;
	int b_mod = 0;
	if (a == 1)
	{
		print("This number is prime");
	}
	if (a<1)
	{
		print("A number greater than 0 please");
	}
	if (a>1)	
	{
		while (counter <= current_a)
		{
			b_mod = current_a % counter;
			if (b_mod ==0)	
			{
				if (counter != a)
				{
					prime = 0;
					print(counter);
					current_a = current_a / counter;
				}
				else
				{
					counter = counter+1;
				}
			}
			else
			{
				counter = counter+1;	
			}
		}
		
		if (prime==1)
		{	
			print("it's prime");
		}
	}
	
}
primeNumberChecker(5);
primeNumberChecker(27);
primeNumberChecker(43);