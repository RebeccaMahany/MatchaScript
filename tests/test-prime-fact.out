Current number:
5
it's prime
Current number:
27
3
3
3
Current number:
43
it's prime
